sdk/identity/identity/src/credentials/azurePipelinesServiceConnectionCredential.ts

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AccessToken, GetTokenOptions, TokenCredential } from "@azure/core-auth";
+import { ClientAssertionCredential } from "./clientAssertionCredential";
+import { CredentialUnavailableError } from "../errors";
+import { credentialLogger } from "../util/logging";
+import { checkTenantId } from "../util/tenantIdUtils";
+import {
+  createDefaultHttpClient,
+  createHttpHeaders,
+  createPipelineRequest,
+} from "@azure/core-rest-pipeline";
+import { AzurePipelinesServiceConnectionCredentialOptions } from "./azurePipelinesServiceConnectionCredentialOptions";
+
+const credentialName = "AzurePipelinesServiceConnectionCredential";
+
+const logger = credentialLogger(credentialName);
+
+export class AzurePipelinesServiceConnectionCredential implements TokenCredential {
+  private client: ClientAssertionCredential | undefined;
+  private serviceConnectionId: string | undefined;
+
+  /**
+   * AzurePipelinesServiceConnectionCredential supports Microsoft Entra Workload ID on Kubernetes.
+   *
+   * @param options - The identity client options to use for authentication.
+   */
+  constructor(
+    clientId: string,
+    tenantId: string,
+    serviceConnectionId: string,
+    options?: AzurePipelinesServiceConnectionCredentialOptions
+  ) {
+    checkTenantId(logger, tenantId);
```

Comment:
Should we throw if clientId is nullish? Same for tenantId?

Reply:
I think the constructor won't allow a null to be passed right?

Reply:
Also tenantId validation is done by `checkTenantId` function

Reply:
Yeah 100% - for TypeScript users the type system will not allow null / undefined values there. But I think for JS users we should still have runtime validation. What do you think? Do other credentials throw if clientId or tenantid are nullish?



> Also tenantId validation is done by `checkTenantId` function

Right, that makes sense! The error message in that function is pretty clear as well. So just clientId should be validated I suppose

Reply:
Not blocking this PR if you prefer to make an issue and get a dev-build going